services/skus/controllers.go

```diff
@@ -1064,151 +1066,139 @@ func HandleRadomWebhook(service *Service) handlers.AppHandler {
-				if err != nil {
-					sublogger.Error().Err(err).Msg("failed to update order metadata")
+
+				if err := service.Datastore.AppendOrderMetadata(ctx, &orderID, "stripeSubscriptionId", subscription.ID); err != nil {
+					lg.Error().Err(err).Msg("failed to update order metadata")
 					return handlers.WrapError(err, "error updating order metadata", http.StatusInternalServerError)
 				}
-				// set paymentProcessor as stripe
-				err = service.Datastore.AppendOrderMetadata(ctx, &orderID, paymentProcessor, StripePaymentMethod)
-				if err != nil {
-					sublogger.Error().Err(err).Msg("failed to update order to add the payment processor")
+
+				if err := service.Datastore.AppendOrderMetadata(ctx, &orderID, paymentProcessor, StripePaymentMethod); err != nil {
+					lg.Error().Err(err).Msg("failed to update order to add the payment processor")
 					return handlers.WrapError(err, "failed to update order to add the payment processor", http.StatusInternalServerError)
 				}
 
-				sublogger.Debug().Str("orderID", orderID.String()).Msg("order is now paid")
-				return handlers.RenderContent(r.Context(), "payment successful", w, http.StatusOK)
+				lg.Debug().Str("orderID", orderID.String()).Msg("order is now paid")
+				return handlers.RenderContent(ctx, "payment successful", w, http.StatusOK)
 			}
 
-			sublogger.Debug().
-				Str("orderID", orderID.String()).Msg("order not paid, set pending")
-			err = service.Datastore.UpdateOrder(orderID, "pending")
-			if err != nil {
-				sublogger.Error().Err(err).Msg("failed to update order status")
+			lg.Debug().Str("orderID", orderID.String()).Msg("order not paid, set pending")
+
+			if err := service.Datastore.UpdateOrder(orderID, "pending"); err != nil {
+				lg.Error().Err(err).Msg("failed to update order status")
 				return handlers.WrapError(err, "error updating order status", http.StatusInternalServerError)
 			}
-			sublogger.Debug().
-				Str("sub_id", subscription.ID).Msg("set subscription id in order metadata")
-			err = service.Datastore.AppendOrderMetadata(ctx, &orderID, "stripeSubscriptionId", subscription.ID)
-			if err != nil {
-				sublogger.Error().Err(err).Msg("failed to update order metadata")
+
+			lg.Debug().Str("sub_id", subscription.ID).Msg("set subscription id in order metadata")
+
+			if err := service.Datastore.AppendOrderMetadata(ctx, &orderID, "stripeSubscriptionId", subscription.ID); err != nil {
+				lg.Error().Err(err).Msg("failed to update order metadata")
 				return handlers.WrapError(err, "error updating order metadata", http.StatusInternalServerError)
 			}
-			sublogger.Debug().
-				Str("sub_id", subscription.ID).Msg("set ok response")
-			return handlers.RenderContent(r.Context(), "payment failed", w, http.StatusOK)
-		}
 
-		// Handle subscription cancellations
-		if event.Type == StripeCustomerSubscriptionDeleted {
+			lg.Debug().Str("sub_id", subscription.ID).Msg("set ok response")
```

Comment:
question: do you think we could remove the debug logging?

Not sure how much the debugging adds over and above the error logging

Reply:
I would very much love to.

Reply:
Updated.